lib/core/src/sdk.rs

```diff
@@ -2309,211 +2307,115 @@ impl LiquidSdk {
-            match update_swap_info_res {
-                Ok((PaymentState::Recoverable, PaymentState::Created))
-                | Ok((PaymentState::Recoverable, PaymentState::Pending)) => {
-                    info!("Tracking recoverable swap: {swap_id}");
-                    if let Err(e) = self.status_stream.track_swap_id(&swap_id) {
-                        error!("Unable to track swap id {swap_id}: {e}");
-                    }
-                }
-                _ => {}
-            }
-        }
+        for swap in recoverable_swaps {
+            let swap_id = &swap.id();
 
-        // Loop over the recovered chain data for Created, Pending or Recoverable send swaps
-        for (swap_id, send_data) in recovered_onchain_data.send {
-            let to_state: PaymentState = send_data
-                .derive_partial_state()
-                .unwrap_or(PaymentState::Created);
-            let lockup_tx_id = send_data.lockup_tx_id.clone().map(|h| h.txid.to_string());
-            let refund_tx_id = send_data.refund_tx_id.clone().map(|h| h.txid.to_string());
-            let history_updates = vec![send_data.lockup_tx_id, send_data.refund_tx_id]
-                .into_iter()
-                .flatten()
-                .collect::<Vec<HistoryTxId>>();
-            for history in history_updates {
-                if let Some(tx) = tx_map.remove(&history.txid) {
-                    self.persister
-                        .insert_or_update_payment_with_wallet_tx(&tx)?;
-                }
-            }
-            let update_swap_info_res = self
-                .send_swap_handler
-                .update_swap_info(
-                    &swap_id,
-                    to_state,
-                    None,
-                    lockup_tx_id.as_deref(),
-                    refund_tx_id.as_deref(),
-                )
-                .await;
-            match update_swap_info_res {
-                Ok((PaymentState::Recoverable, PaymentState::Created))
-                | Ok((PaymentState::Recoverable, PaymentState::Pending)) => {
-                    info!("Tracking recoverable swap: {swap_id}");
-                    if let Err(e) = self.status_stream.track_swap_id(&swap_id) {
-                        error!("Unable to track swap id {swap_id}: {e}");
+            let update_swap_info_res = match swap {
+                Swap::Receive(ReceiveSwap {
+                    state,
+                    claim_tx_id,
+                    mrh_tx_id,
+                    ..
+                }) => {
+                    let history_updates = vec![&claim_tx_id, &mrh_tx_id]
+                        .into_iter()
+                        .flatten()
+                        .collect::<Vec<&String>>();
+                    for tx_id in history_updates {
+                        if let Some(tx) =
+                            tx_map.remove(&lwk_wollet::elements::Txid::from_str(tx_id)?)
+                        {
+                            self.persister
```

Comment:
This seems to repeat on every swap for every transaction that is associated with the wallet. 
Isn't it better to add them all by iterating the tx_map right on start?

Reply:
Not sure I've understood. The loop only runs on the "monitored" (recoverable) swaps, and the TxMap is the synced map <txid,tx> which we use to retrieve the data, so it is necessary to load it fully

Reply:
I think this way prevents unneeded updates. We pop any recovered tx's associated to the monitored swaps, then check if the remaining tx's need to be updated at the bottom of the sync
https://github.com/breez/breez-sdk-liquid/blob/5f1fda010009c309d46f14e6b49e7b1ca9325e66/lib/core/src/sdk.rs#L2481-L2503